wp/lib/bap_wp/src/precondition.ml

```diff
@@ -1199,18 +1201,60 @@ let init_vars (vars : Var.Set.t) (env : Env.t) : Constr.t list * Env.t =
         debug "Initializing var: %s\n%!" (Constr.to_string comp);
         comp :: inits, env)
 
+let user_func_spec (sub_name : string) (sub_pre : string) (sub_post : string)
```

Comment:
Wouldn't hurt to have a small comment here as well.

Reply:
You mean like a function summary? I have a description of `user_func_spec` in the `.mli` already, but I can add one here if you think it's a good idea.

Reply:
My rule of thumb is to answer "what does this function do?" in the `.mli` and "how does this function work" in the `.ml`, but I'm not very consistent with that.

Reply:
I added a comment/explanation before `user_func_spec` in `precondition.ml`. I would love to hear your thoughts on it.

Reply:
Looks good to me.